### PR TITLE
feat(argo-cd): Add scrapeTimeout property to server and application…

### PR DESCRIPTION
**Pull request by nikolay-o**
I've added scrapeTimeout property to server and application, applicationSet controllers.

### CODE REVIEW

File: charts/argo-cd/Chart.yaml [hunk 1]
```diff
@@ -28,3 +28,5 @@ annotations:
   artifacthub.io/changes: |
     - kind: changed
       description: Improved documentation for various ingress setups
```
Comment (yu-croco) [inline charts/argo-cd/Chart.yaml:30]: Please remove old one.

Comment (nikolay-o) [reply charts/argo-cd/Chart.yaml]: Fixed, thanks for review

File: charts/argo-cd/templates/argocd-application-controller/servicemonitor.yaml [hunk 2]
```diff
@@ -22,6 +22,7 @@ spec:
       {{- with .Values.controller.metrics.serviceMonitor.interval }}
       interval: {{ . }}
       {{- end }}
+      scrapeTimeout: {{ .Values.controller.metrics.serviceMonitor.scrapeTimeout }}
```
Comment (yu-croco) [inline charts/argo-cd/templates/argocd-application-controller/servicemonitor.yaml:25]: maybe it's better to align to other values, like `interval` ? 🤔  WDYT?
```yaml
      {{- with .Values.controller.metrics.serviceMonitor.interval }}
      interval: {{ . }}
      {{- end }}
```

Comment (yu-croco) [reply charts/argo-cd/templates/argocd-application-controller/servicemonitor.yaml:25]: Ah but default value is defined. hmm it doesn't make much difference.

**Review by yu-croco — APPROVED** [review]
LGTM

File: charts/argo-cd/values.yaml [hunk 3]
```diff
@@ -758,6 +758,7 @@ controller:
   metrics:
     # -- Deploy metrics service
     enabled: false
+    scrapeTimeout: 10s
```
Comment (mkilchhofer) [inline charts/argo-cd/values.yaml:761]: Missing docs

File: charts/argo-cd/README.md [hunk 4]
```diff
@@ -727,6 +727,7 @@ NAME: my-release
 | controller.metrics.rules.namespace | string | `""` | PrometheusRule namespace |
 | controller.metrics.rules.selector | object | `{}` | PrometheusRule selector |
 | controller.metrics.rules.spec | list | `[]` | PrometheusRule.Spec for the application controller |
+| controller.metrics.scrapeTimeout | string | `"10s"` |  |
```
Comment (mkilchhofer) [inline charts/argo-cd/README.md:730]: Missing docs

**Review by mkilchhofer — CHANGES_REQUESTED** [review]

File: charts/argo-cd/values.yaml (hunk 3 above)
Comment (nikolay-o) [reply charts/argo-cd/values.yaml:762]: fixed

File: charts/argo-cd/README.md (hunk 4 above)
Comment (nikolay-o) [reply charts/argo-cd/README.md]: fixed

File: charts/argo-cd/templates/argocd-application-controller/servicemonitor.yaml (hunk 2 above)
Comment (mkilchhofer) [reply charts/argo-cd/templates/argocd-application-controller/servicemonitor.yaml:25]: I would also preffer this method and unsetting the default in values.yaml

Prometheus-operator has no default value on ServiceMonitors, the default is inherited from the global config of prom-operator.
Ref: https://github.com/prometheus-community/helm-charts/blob/94330ab7439a5ab2701bbe8605097a904223d62c/charts/kube-prometheus-stack/charts/crds/crds/crd-servicemonitors.yaml#L470-L473

So I suggest to define it like this in values:
```yaml
      # -- (string) Prometheus ServiceMonitor scrapeTimeout. If empty, Prometheus uses the global scrape timeout unless it is less than the target's scrape interval value in which the latter is used.
       scrapeTimeout: ~
```

**Comment by nikolay-o** [comment]
@mkilchhofer I fixed documentation, could you please look?

File: charts/argo-cd/templates/argocd-application-controller/servicemonitor.yaml (hunk 2 above)
Comment (nikolay-o) [reply charts/argo-cd/templates/argocd-application-controller/servicemonitor.yaml]: @mkilchhofer that's make sense, replaced

**Review by mkilchhofer — APPROVED** [review]

**Comment by mkilchhofer** [comment]
> @mkilchhofer I fixed documentation, could you please look?

Perfect, thank you so much for your contribution.
Lgtm👍

**Review by mbevc1 — APPROVED** [review]

**Pull request merged**